pkg/sidecar/proxy/proxy_test.go

```diff
@@ -61,14 +61,15 @@ var _ = Describe("Reverse Proxy", func() {
 				proxy := NewProxy("0", targetURL, cfg) // port 0 to automatically choose one that's available.
 
 				ctx, cancelFn := context.WithCancel(ctx)
-				defer cancelFn()
+				stoppedCh := make(chan bool)
```

Comment:
nit: if you'll never send `false` down the channel then no need for a bool. Can use `chan struct{}` if only interested in read-readiness.